### PR TITLE
MAINT: statespace cleanup, alphabetically

### PR DESCRIPTION
**Pull request by jbrockmendel**
@ChadFulton this is getting files alphabetically from the top, whereas the others are going from the bottom; should be non-overlapping.

### CODE REVIEW

File: statsmodels/tsa/statespace/tests/test_initialization.py [hunk 1]
```diff
@@ -374,34 +369,54 @@ def test_nested():
...
     # Invalid initializations (also tests for some invalid calls to set)
-    assert_raises(ValueError, Initialization, 5, '')
-    assert_raises(ValueError, Initialization, 5, 'stationary', constant=[1, 2])
-    assert_raises(ValueError, Initialization, 5, 'stationary',
-                  stationary_cov=[1, 2])
-    assert_raises(ValueError, Initialization, 5, 'known')
-    assert_raises(ValueError, Initialization, 5, 'known', constant=[1])
-    assert_raises(ValueError, Initialization, 5, 'known', stationary_cov=[0])
+    with pytest.raises(ValueError):
+        Initialization(5, '')
```
Comment (josef-pkt) [inline statsmodels/tsa/statespace/tests/test_initialization.py]: I don't like pytest for simple assert_raises
but here it is Chad's code.

File: statsmodels/tsa/statespace/tests/test_exact_diffuse_filtering.py [hunk 2]
```diff
@@ -337,7 +334,7 @@ def test_results(self):
         assert_allclose(res.predicted_state_cov[:, :, 3], P4)
 
         # Miscellaneous
-        assert_equal(res.nobs_diffuse, 3)
+        assert res.nobs_diffuse == 3
```
[the remark this answers is not in the document]
Comment (jbrockmendel) [reply statsmodels/tsa/statespace/tests/test_exact_diffuse_filtering.py]: Um... how come?  in the other thread you said you were allowing it now.  What's wrong with making this clearer to readers and using our tools as intended?

File: statsmodels/tsa/statespace/tests/test_initialization.py (hunk 1 above)
Comment (jbrockmendel) [reply statsmodels/tsa/statespace/tests/test_initialization.py]: The contextmanager version is much friendlier for copy/pasting when running in the REPL, plus readability.

> but here it is Chad's code.

Yep, I can point you towards some more trivial PRs in non-Chad's code if that would help.

**Comment by josef-pkt** [comment]
I still prefer numpy.testing asserts to pytest, when numpy's assert are appropriate for the case.
I still don't trust the pytest "magic".  
(i.e. I can easily check the code that numpy.testing uses and it has worked for us for a long time, I have no idea how pytest's behind the scene rewriting works, nor where to quickly look it up.)

File: statsmodels/tsa/statespace/tests/test_exact_diffuse_filtering.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (josef-pkt) [reply statsmodels/tsa/statespace/tests/test_exact_diffuse_filtering.py]: No, I said that I led a few of those go through, for example in a previously "weird" assert of an `isinstance` check.

Equality and almost equality testing of numbers are definitely the domain of numpy and not of pytest and will always stay that way (as long as numpy supports it).
And I'm not in favor of wholesale rewriting of asserts until I'm more convinced that pytest doesn't hurt.

**Comment by josef-pkt** [comment]
(just to make the point in general, again:
I didn't trust pip and pytest, and to some extend still don't, because those developers have a completely different background and preferences than we, or I, do. In contrast, I know that numpy and scipy developers have largely the same preferences and use cases as us. 
However, this has started to change given the large proportion of python usage in the "data science" area which creates a counterweight to, for example, the web development use cases.
)

**Comment by jbrockmendel** [comment]
You’ve done a good job today merging easy things elsewhere.  Keep up the good work.

**Pull request closed without merging**